Lib/unittest/mock.py

```diff
@@ -2173,7 +2173,15 @@ def __init__(self, /, *args, **kwargs):
         self.__dict__['_mock_await_args'] = None
         self.__dict__['_mock_await_args_list'] = _CallList()
         code_mock = NonCallableMock(spec_set=CodeType)
-        code_mock.co_flags = inspect.CO_COROUTINE
+        code_mock.co_flags = (
```

Comment:
Maybe it would be better to use a donor async function and copy the co_flags from that?

Reply:
Do you have some existing one in mind? 🤔

Reply:
```python
async def _donor(*args, **kwargs):
    pass

ASYNC_DEF_FLAGS = _donor.__code__.co_flags
```

Something like that?

Reply:
```python
>>> ASYNC_DEF_FLAGS
143
```

This means: `COROUTINE` + `VARKEYWORDS` `VARARGS` + `NEWLOCALS` + `OPTIMIZED`

Docs:

```
.. data:: CO_OPTIMIZED

   The code object is optimized, using fast locals.

.. data:: CO_NEWLOCALS

   If set, a new dict will be created for the frame's ``f_locals`` when
   the code object is executed.
```

I am not sure this is correct 🤔

Reply:
Would be good to get @lisroach's view on this...